parser/src/json/schema.rs

```diff
@@ -126,6 +128,43 @@ impl Schema {
         }
     }
 
+    pub fn const_compile(&self) -> Option<RegexNode> {
+        let str = match self {
+            Schema::Null => "null",
+            Schema::Boolean => return Some(RegexNode::Regex("true|false".to_string())),
+            Schema::Number {
+                minimum: Some(x),
+                maximum: Some(y),
+                ..
+            } if x == y => return Some(RegexNode::Literal(x.to_string())),
+            Schema::String {
+                // TODO: check that min/max don't break the literal and all the other constraints
+                // or just convert this whole node to a regex
+                min_length: 0,
+                max_length: None,
+                regex: Some(RegexAst::Literal(s)),
+            } => &regex_syntax::escape(&serde_json::to_string(s).unwrap()),
```

Comment:
This is the bit that makes it a bit tricky to do this for general string regexes -- json quoting

Reply:
I'll add AST node to escape. Shouldn't be hard.